scene/main/viewport.cpp

```diff
@@ -2828,16 +2828,26 @@ void Viewport::_post_gui_grab_click_focus() {
 
 ///////////////////////////////
 
-void Viewport::push_text_input(const String &p_text) {
+void Viewport::_push_text_input(const String &p_text, bool p_emit_text_changed_signal) {
 	ERR_MAIN_THREAD_GUARD;
 	if (gui.subwindow_focused) {
 		gui.subwindow_focused->push_text_input(p_text);
 		return;
 	}
 
-	if (gui.key_focus) {
-		gui.key_focus->call("set_text", p_text);
+	StringName set_text_method = p_emit_text_changed_signal
+			? SNAME("_set_text_and_emit_text_changed_signal")
+			: SNAME("set_text");
```

Comment:
Since you are adding a new method anyway, you could make it handle both emitting and not emitting the signal (it can be a parameter). And the method could be renamed to something that isn't an implementation detail, like `_push_text_input()`.

Reply:
You mean something like `{Line,Text}Edit::_push_text_input(const String &p_text, bool p_emit_signal)`?